pkg/cert/v1beta1/generate/generate_test.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2021 The Kubeflow Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generate
+
+import (
+	"github.com/kubeflow/katib/pkg/cert/v1beta1/consts"
+	admissionregistration "k8s.io/api/admissionregistration/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"log"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"testing"
+)
+
+func TestGenerate(t *testing.T) {
+
+	const testNamespace = "test"
+
+	testGeneratorJob := &batchv1.Job{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "batch/v1",
+			Kind:       "Job",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      consts.JobName,
+			Namespace: testNamespace,
+			UID:       "test",
+		},
+	}
+	testValidatingWebhook := &admissionregistration.ValidatingWebhookConfiguration{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "admissionregistration.k8s.io/v1",
+			Kind:       "ValidatingWebhookConfiguration",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "katib.kubeflow.org",
+		},
+		Webhooks: []admissionregistration.ValidatingWebhook{
+			{
+				Name: "validator.experiment.katib.kubeflow.org",
+				ClientConfig: admissionregistration.WebhookClientConfig{
+					CABundle: []byte("CG=="),
+				},
+			},
+		},
+	}
+	testMutatingWebhook := &admissionregistration.MutatingWebhookConfiguration{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "admissionregistration.k8s.io/v1",
+			Kind:       "MutatingWebhookConfiguration",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "katib.kubeflow.org",
+		},
+		Webhooks: []admissionregistration.MutatingWebhook{
+			{
+				Name: "defaulter.experiment.katib.kubeflow.org",
+				ClientConfig: admissionregistration.WebhookClientConfig{
+					CABundle: []byte("CG=="),
+				},
+			},
+			{
+				Name: "mutator.pod.katib.kubeflow.org",
+				ClientConfig: admissionregistration.WebhookClientConfig{
+					CABundle: []byte("CG"),
+				},
+			},
+		},
+	}
+
+	oldWebhookCertSecret := &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      consts.Secret,
+			Namespace: testNamespace,
+		},
+	}
+
+	tests := []struct {
+		name      string
```

Comment:
Let's keep the name `testDescription` to be consistent with other ut:
https://github.com/kubeflow/katib/blob/master/pkg/controller.v1beta1/suggestion/composer/composer_test.go#L147
https://github.com/kubeflow/katib/blob/master/pkg/controller.v1beta1/suggestion/suggestionclient/suggestionclient_test.go#L192

Does it sound good @tenzen-y ?

Reply:
Sure, sounds good!